code/datums/components/riding.dm

```diff
@@ -50,6 +54,32 @@
 	M.set_glide_size(movable_parent.glide_size)
 	M.updating_glide_size = FALSE
 	handle_vehicle_offsets(movable_parent.dir)
+	if(can_use_abilities)
+		setup_abilities(M)
+
+///Gives the rider the riding parent's abilities
+/datum/component/riding/proc/setup_abilities(mob/living/M)
+
+	if(!istype(parent, /mob/living))
+		return
+
+	var/mob/living/ridden_creature = parent
+
+	for(var/i in ridden_creature.abilities)
+		var/obj/effect/proc_holder/proc_holder = i
+		M.AddAbility(proc_holder)
```

Comment:
by the way, spells don't count as abilities, right? if I somehow get a wizard into the body of a vatbeast, I can't copy his spells, right?

Reply:
dunno dont think so